tests/NSubstitute.Acceptance.Specs/ArgumentMatching.cs

```diff
@@ -346,6 +346,17 @@ public void Should_fail_with_redundant_exceptions_if_arg_matchers_misused()
             });
         }
 
+        public delegate string DelegateWithOutParameter(string input, out string result);
+
+        [Test]
+        public void Should_recognize_out_parameters_for_delegate_and_match_specification()
+        {
+            var subs = Substitute.For<DelegateWithOutParameter>();
+
+            Assert.DoesNotThrow(() =>
+                subs.Invoke(Arg.Any<string>(), out var _).Returns("42"));
```

Comment:
I would prefer this test did a quick check of the substitute rather than just asserting it can be constructed.
Something like:

```csharp
sub.Invoke(Arg.Any<string>(), out var _).Returns("42");
var result = sub.Invoke("test", out var output);
Assert.Equals("42", result);
Assert.Equals(null, output); // or whatever output we expect ;)
```

Reply:
Totally makes sense. Changed the test the result.
I don't configure the `out` parameter value, therefore decided to not test it.